### ISSUE TITLE
Imports through ISBN searches create duplicate works

### ISSUE THREAD

**Issue by seabelis** [internetarchive/openlibrary#3473]
When a user searches by ISBN and the edition is not already present in the catalog, the edition is imported. However, this frequently creates a new work when an appropriate work already exists.

### Relevant url?
https://openlibrary.org/books/OL28122354M/The_Hound_of_the_Baskervilles (imported edition)
https://openlibrary.org/works/OL20779126W/The_Hound_of_the_Baskervilles (created work)
This import should have been associated with https://openlibrary.org/works/OL262454W/The_Hound_of_the_Baskervilles

### Steps to Reproduce

1. Search for an ISBN that does not exist in the Open Library catalog, but exists on amazon and has a corresponding work on Open Library.
2. If imported, check to see if a new work (duplicate) was created or if the edition is associated with the existing work. (I've experienced both cases.)

* Actual: Sometimes the import creates a new work when an appropriate work already exists.
* Expected: If an appropriate work exists, the import should not create a new work.

### Details

- **Logged in (Y/N)?** 
- **Browser type/version?** 
- **Operating system?** 
- **Environment (prod/dev/local)?** prod

### Stakeholders

@mekarpeles @cdrini

**Comment by mekarpeles** [comment]
@seabelis I think we need an example that remains unmerged :)

**Comment by jzacsh** [comment]
I think this just happened to me, so I've purposely not merged them (hopefully they stay that way while you investigate):
- a) https://openlibrary.org/works/OL20791728W/Lily_and_the_octopus
- b) https://openlibrary.org/works/OL20040818W/Lily_and_the_octopus

## some hopefully helpful details:
1. these are both duplicates (i'm _guessing_ they came into being via my goodreads csv import where i had an ISBN `1501126229` and `9781501126222` present<sup>1</sup>)
2. they're *both* attached correctly attached to this author: https://openlibrary.org/authors/OL7611067A/Steven_Rowley
3. `works/OL20791728W` is the one where my data shows up, so I'm _guessing_ that's perhaps the duplicate generated by my CSV.
4. interestingly (perhaps worth separate issue-filing to discuss?) it's surprisingly difficult to spot/fix/understand duplicate works<sup>2</sup>
   So questions this raised for me to bring to this issue tracker:
   - 4a) is there a way to _detect_ duplicates?
   - 4b) when someone _does_ find a dupe, how do you _report/fix_ it?
     Currently the top search result i found was this FAQ https://openlibrary.org/help/faq?v=102#delete-book says to email info@openlibrary.org but i'm guessing that's not intended/scalable way to merge duplicates.
   - 4c) is there an author-page bug here causing both works not to list?
      **Concretely**: if you click the author's link from either work, you can't find your way to the other work from the author's page (despite point 2 above). Once you land on the author's page, only `works/OL20040818W` shows.

----

<sup>1</sup>: here's the relevant snippet of my CSV:
```
Book Id,Title,Author,Author l-f,Additional Authors,ISBN,ISBN13,My Rating,Average Rating,Publisher,Binding,Number of Pages,Year Published,Original Publication Year,Date Read,Date Added,Bookshelves,Bookshelves with positions,Exclusive Shelf,My Review,Spoiler,Private Notes,Read Count,Recommended For,Recommended By,Owned Copies,Original Purchase Date,Original Purchase Location,Condition,Condition Description,BCID
27276262,Lily and the Octopus,Steven  Rowley,"Rowley, Steven",,"=""1501126229""","=""9781501126222""",5,3.71,Simon  Schuster,Hardcover,307,2016,2016,2018/04/16,2018/02/25,own-hardcopy,own-hardcopy (#3),read,"I read over half the book in just a day, without realizing it. Beware: you will bawl.",,,1,,,1,,,unspecified,,
```
<sup>2</sup>: I was only able to do it because I knew I should have had _"already read"_ data and a rating attached, but I wasn't seeing it - finding the other edition required my manually paging through my profile's data until i found a link to the book and indeed it was a different work ID)

**Comment by jzacsh** [comment]
i'm guessing search index updated and now there's actually _four_ duplicates of the same work (see screenshot):
1. [`/works/OL17356798W`](https://openlibrary.org/works/OL17356798W)
2. [`/works/OL20040818W`](https://openlibrary.org/works/OL20040818W)
3. [`/works/OL21320871W`](https://openlibrary.org/works/OL21320871W)
4. [`/works/OL20791728W`](https://openlibrary.org/works/OL20791728W)

![Screenshot 2020-12-20 at 13 06 32](https://user-images.githubusercontent.com/156228/102722072-498e0700-42c4-11eb-8287-0090d012e539.png)

**Comment by jzacsh** [comment]
okay last update, in case i'm just causing noise for veteran OL folks :)

i'm going to leave this for mek and others familiar with OL to sort out, but as a **newbie** to OL (ie: possibly very wrong) here's what i've understood so far:

## problem: import creates new work for any unrecognized isbn (ie: edition)

I _think_ the problem is that the import logic searches against an ISBN, and if that edition doesn't exist in OL already, the API returns a **newly created edition** and correspondingly a **newly created work**. it's the newly created work which is perhaps the only bug?

## solution: make best-effort to match against works by non-ID means (eg: title,author, etc.)

if that^ understanding is right, then one fix could be: a bit of new UX and wiring into the (solr?) search; eg:
1. update the import logic, when it encounters an unrecognized ISBN, to first do a general search against `(title,author)` (as happens under the hood by users, [eg search above](https://github.com/internetarchive/openlibrary/issues/3473#issuecomment-748647916))).
2. If there's a match against said search, include that in the import UI but let users see that we're doing two things:
   - a) we're proposing creating a whole new edition
   - b) we're proposing that this is actually part of `/works/1234`

by adding (2b) user can untick the row if it's wrong, and/or we can add a step (3) that let's the user correct it inline with some new UX.
